### PR TITLE
Added upgrade for v3.2.0,v3.2.1 and v3.2.2

### PR DESCRIPTION
**Pull request by anpanigr**
Added Operator Version Upgrade for v3.2.0/3.2.1 

WebLogic Domain 

testOperatorWlsUpgradeFrom260ToMain() 
testOperatorWlsUpgradeFrom303ToMain()
testOperatorWlsUpgradeFrom304ToMain()
testOperatorWlsUpgradeFrom313ToMain()
testOperatorWlsUpgradeFrom314ToMain()
testOperatorWlsUpgradeFrom320ToMain()
testOperatorWlsUpgradeFrom321ToMain()
testOperatorWlsUpgradeFrom322ToMain()

FMW Domain 
testOperatorFmwUpgradeFrom260ToMain() 
testOperatorFmwUpgradeFrom303ToMain() 
testOperatorFmwUpgradeFrom304ToMain() 
testOperatorFmwUpgradeFrom313ToMain() 
testOperatorFmwUpgradeFrom314ToMain() 
testOperatorFmwUpgradeFrom320ToMain() 
testOperatorFmwUpgradeFrom321ToMain()
testOperatorFmwUpgradeFrom322ToMain()

( Sequential run for Upgrade Test ) 
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4855
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4875

### CODE REVIEW

**Review by rjeberhard — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java [hunk 1]
```diff
@@ -231,11 +231,7 @@ public void testOperatorFmwUpgradeFrom303ToDevelop() {
...
   /**
-   * Operator upgrade from 3.0.4 to latest.
-   * Install 3.0.4 Operator from GitHub chart repository and create a domain.
-   * Deploy an application to the cluster in domain and verify the application can be
-   * accessed while the operator is upgraded and after the upgrade.
-   * Upgrade operator with latest Operator image and verify CRD version and image are updated.
+   * Operator upgrade from 3.0.4 to latest with a FMW Domain.
    */
   @Test
   @DisplayName("Upgrade Operator from 3.0.4 to develop")
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java:237]: Upgrade Operator from 3.0.4 to main

**Review by xiancao — COMMENTED** [review]
Since main branch is our default branch now, we will upgrade to main instead of develop branch. We may need to update all the comments and method names to reflect that.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java [hunk 2]
```diff
@@ -206,9 +210,9 @@ public void tearDown() {
...
   /**
-   * Operator upgrade from 2.6.0 to latest.
-   * Install 2.6.0 release Operator from GitHub chart repository and create a FMW domain.
-   * Delete Operator and install latest Operator and verify CRD version is updated.
+   * Operator upgrade from 2.6.0 to latest with a FMW Domain.
+   * Delete Operator and install latest Operator.
+   * Verify Domain resource version is updated while domain is in running state.
    */
   @Test
   @DisplayName("Upgrade Operator from 2.6.0 to develop")
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java [hunk 3]
```diff
@@ -217,11 +221,7 @@ public void testOperatorFmwUpgradeFrom260ToDevelop() {
...
   /**
-   * Operator upgrade from 3.0.3 to latest.
-   * Install 3.0.3 Operator from GitHub chart repository and create a domain.
-   * Deploy an application to the cluster in domain and verify the application can be
-   * accessed while the operator is upgraded and after the upgrade.
-   * Upgrade operator with latest Operator image and verify CRD version and image are updated.
+   * Operator upgrade from 3.0.3 to latest with a FMW Domain.
    */
   @Test
   @DisplayName("Upgrade Operator from 3.0.3 to develop")
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java [hunk 4]
```diff
@@ -245,29 +241,39 @@ public void testOperatorFmwUpgradeFrom304ToDevelop() {
...
-   * Deploy an application to the cluster in domain and verify the application can be
-   * accessed while the operator is upgraded and after the upgrade.
-   * Upgrade operator with latest Operator image and verify CRD version and image are updated.
+   * Operator upgrade from 3.1.3 to latest with a FMW Domain.
    */
   @Test
-  @DisplayName("Upgrade Operator from 3.1.2 to develop")
-  public void testOperatorFmwUpgradeFrom312ToDevelop() {
-    installAndUpgradeOperator("3.1.2", DEFAULT_EXTERNAL_SERVICE_NAME_SUFFIX, true);
+  @DisplayName("Upgrade Operator from 3.1.3 to develop")
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java [hunk 5]
```diff
@@ -245,29 +241,39 @@ public void testOperatorFmwUpgradeFrom304ToDevelop() {
...
+  public void testOperatorFmwUpgradeFrom313ToDevelop() {
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java [hunk 6]
```diff
@@ -245,29 +241,39 @@ public void testOperatorFmwUpgradeFrom304ToDevelop() {
...
-   * Deploy an application to the cluster in domain and verify the application can be
-   * accessed while the operator is upgraded and after the upgrade.
-   * Upgrade operator with latest Operator image and verify CRD version and image are updated.
+   * Operator upgrade from 3.1.4 to latest with a FMW Domain.
    */
   @Test
-  @DisplayName("Upgrade Operator from 3.1.3 to develop")
-  public void testOperatorFmwUpgradeFrom313ToDevelop() {
-    installAndUpgradeOperator("3.1.3", DEFAULT_EXTERNAL_SERVICE_NAME_SUFFIX, true);
+  @DisplayName("Upgrade Operator from 3.1.4 to develop")
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java [hunk 7]
```diff
@@ -195,6 +186,28 @@ public void testOperatorWlsUpgradeFrom314ToDevelop(String domainType) {
     upgradeOperator(domainType, "3.1.4", DEFAULT_EXTERNAL_SERVICE_NAME_SUFFIX, true);
   }
 
+  /**
+   * Operator upgrade from 3.2.0 to latest.
+   */
+  @ParameterizedTest
+  @DisplayName("Upgrade Operator from 3.2.0 to develop")
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java [hunk 8]
```diff
@@ -195,6 +186,28 @@ public void testOperatorWlsUpgradeFrom314ToDevelop(String domainType) {
...
+  public void testOperatorWlsUpgradeFrom320ToDevelop(String domainType) {
+    logger.info("Starting test testOperatorWlsUpgradeFrom320ToDevelop with domain type {0}", domainType);
+    upgradeOperator(domainType, "3.2.0", DEFAULT_EXTERNAL_SERVICE_NAME_SUFFIX, true);
+  }
+
+  /**
+   * Operator upgrade from 3.2.1 to latest.
+   */
+  @ParameterizedTest
+  @DisplayName("Upgrade Operator from 3.2.1 to develop")
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java [hunk 9]
```diff
@@ -195,6 +186,28 @@ public void testOperatorWlsUpgradeFrom314ToDevelop(String domainType) {
...
+  @ValueSource(strings = { "domain-in-image", "model-in-image" })
+  public void testOperatorWlsUpgradeFrom321ToDevelop(String domainType) {
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java (hunk 1 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java]: done

**Review by xiancao — APPROVED** [review]
LG

**Pull request merged**